include/revng/ADT/CompilationTime.h

```diff
@@ -200,4 +200,34 @@ template<auto &V>
              or std::is_function_v<std::remove_pointer_t<decltype(V)>>
 using FunctionTraits = detail::FunctionTraits<decltype(V)>;
 
+/// Helper function that converts a pack of booleans to their constexpr
+/// counterpart, this is useful to e.g. have a type depending on a command-line
+/// option. Do note that the compiler will expand this to all the possible
+/// combinations, so given N booleans there will be 2**N template expansions.
+template<typename CallableT, typename... BoolType>
+  requires(std::is_same_v<BoolType, bool> and ...)
+inline constexpr auto
+bools(CallableT &&Callable, bool Bool, BoolType... BoolRest) {
```

Comment:
1. You are an evil person. But I like the way you think.
2. I'd call this `invokeAllCombinations` (or is it *permutations*?).